### ISSUE TITLE
[Lecture Topic Task] Add making implicit concepts explicit to the documentation

### ISSUE THREAD

**Issue by AnthonyMed13** [uprm-inso4116-2024-2025-s1/semester-project-multiplayer-dominoes#121]
The team describes at least three examples of how they started with an understanding of the domain which they later found had a hidden concept and after some discussion they found the missing concept and made it explicit. They describe how this changes the ubiquitous language of their project.

Temporary Points:
Urgency: 4
Difficulty: 2

**Comment by AnthonyMed13** [comment]
Game Rules and Variations: Initially, we assumed that dominoes had a single set of rules. However, after discussions with stakeholders and between others we realized that many players are accustomed to different rule sets based on their backgrounds. This led to the team explicitly incorporating the idea of the Rule Variations, instead of just referring it to rules in general. The standard ruleset of the game would be the one we traditionally play in Puerto Rico. With this we can now communicate the different rules and need for supporting different rule variations.

Different Types of Players: Initially, we thought of the players as being one as a simple user who played the domino game. As we kept on adding more features and testing different things we found the need to add distinction between the different types of Players. We found that using Random Players to describe anyone you join or joins you in an online lobby was a good way to differentiate between you as the Player or other types of Players added later on in the future. This lets us communicate better about the matchmaking and lobby creation while having a clearer idea on which types of Players we are expecting.

Tutorials and Learning Curves: At the beginning we visualize the need for tutorials as something needed in order to have a bigger audience of people join and play our game. We wanted it to be as accessible as possible for people with experience or no experience with the game. As we tested it, we found out that different people had different learning curves and would pick up on things faster or slower than others. We found the need to expand the initial concept of tutorials and add Learning Curves to further explain the different levels of tutorials needed. With this we can now have a clearer understanding on how our tutorials impact players and let them learn at their own pace.

By exposing these hidden concepts and making them more explicit we found better ways to communicate with the team and stakeholders on decisions that may further improve the current experience for everybody. 

@AngelCIICMorales @Keiven-Soto @ebrain-mendoza121
